### PR TITLE
Update CONTRIBUTING for project's new home

### PR DESCRIPTION
**Pull request by michaelsembwever**
and update new contribution agreements and CLAs signed.

**Commit message must be updated before merging.**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,21 +6,22 @@ This guide outlines the process of landing patches in gocql and the general appr
...
 
-* You should raise a pull request to gocql/gocql on Github
+* You should raise a pull request to apache/cassandra-gocql-driver on Github
 * The pull request has a title that clearly summarizes the purpose of the patch
 * The motivation behind the patch is clearly defined in the pull request summary
-* Your name and email have been added to the `AUTHORS` file (for copyright purposes)
+* You agree that your contribution is donated to the Apache Software Foundation (appropriate copyright is on all new files)
 * The patch will merge cleanly
 * The test coverage does not fall below the critical threshold (currently 64%) 
 * The merge commit passes the regression test suite on Travis
```
Comment (martin-sucha) [inline CONTRIBUTING.md]: We no longer have Travis builds, the current code uses Github Actions to run the tests.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,21 +6,22 @@ This guide outlines the process of landing patches in gocql and the general appr
...
 The following is a check list of requirements that need to be satisfied in order for us to merge your patch:
 
-* You should raise a pull request to gocql/gocql on Github
+* You should raise a pull request to apache/cassandra-gocql-driver on Github
 * The pull request has a title that clearly summarizes the purpose of the patch
 * The motivation behind the patch is clearly defined in the pull request summary
-* Your name and email have been added to the `AUTHORS` file (for copyright purposes)
+* You agree that your contribution is donated to the Apache Software Foundation (appropriate copyright is on all new files)
 * The patch will merge cleanly
 * The test coverage does not fall below the critical threshold (currently 64%)
```
Comment (martin-sucha) [inline CONTRIBUTING.md]: Current coverage seems to be 50.4% of statements :disappointed: 
We should probably update the number here and open an issue to display the coverage in pull requests as a comment (perhaps even enforce the coverage rule in a CI check).

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -6,21 +6,22 @@ This guide outlines the process of landing patches in gocql and the general appr
...
 * The pull request has a title that clearly summarizes the purpose of the patch
 * The motivation behind the patch is clearly defined in the pull request summary
-* Your name and email have been added to the `AUTHORS` file (for copyright purposes)
+* You agree that your contribution is donated to the Apache Software Foundation (appropriate copyright is on all new files)
 * The patch will merge cleanly
 * The test coverage does not fall below the critical threshold (currently 64%) 
 * The merge commit passes the regression test suite on Travis
 * `go fmt` has been applied to the submitted code
 * Notable changes (i.e. new features or changed behavior, bugfixes) are appropriately documented in CHANGELOG.md, functional changes also in godoc
+* A correctly formatted commit message, see https://cassandra.apache.org/_/development/how_to_commit.html#tips
```
Comment (martin-sucha) [inline CONTRIBUTING.md]: It seems to me that it would be helpful to expand the guidance on writing commit messages to a separate section.

Specifically, to say that the description of the commit should explain why the change is necessary

Should we encourage contributors to use signed git commits?

File: NOTICE [hunk 4]
```diff
@@ -48,7 +48,7 @@ Matt Robenolt <mattr@disqus.com>
 Phillip Couto <phillip.couto@stemstudios.com> *
 Niklas Korz <korz.niklask@gmail.com>
 Nimi Wariboko Jr <nimi@channelmeter.com>
-Ghais Issa <ghais.issa@gmail.com>
+Ghais Issa <ghais.issa@gmail.com> *
```
Comment (martin-sucha) [inline NOTICE]: Technically, changes in this file are not related to `CONTRIBUTING.md`, I would prefer if it was a separate commit.

**Review by martin-sucha — COMMENTED** [review]
Thank you for the pull request! I've added some comments inline.

File: CONTRIBUTING.md (hunk 3 above)
Comment (michaelsembwever) [reply CONTRIBUTING.md]: > Should we encourage contributors to use signed git commits?

This isn't project precedence, though it can always be recommended (and should be).
Ultimately, it's not my call.

File: NOTICE (hunk 4 above)
Comment (michaelsembwever) [reply NOTICE]: done, in https://github.com/apache/cassandra-gocql-driver/pull/1792

Comment (martin-sucha) [reply NOTICE]: Thank you! There are still two other entries in NOTICE changed by this pull request.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -6,24 +6,39 @@ This guide outlines the process of landing patches in gocql and the general appr
...
+## Commit Message
+
+The Apache Cassandra project has a commit message precendence like
+```
+<One sentence description, usually Jira title or CHANGES.txt summary>
+
+ patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####
+```
+
+The 'patch by …; reviewed by' line is important.  If enforces our review-than-commit procedure, allowing commits from non-git-branch patches, and is parsed to build the project contribulyse statistics found [here](https://nightlies.apache.org/cassandra/devbranch/misc/contribulyze/html/).
```
Comment (martin-sucha) [inline CONTRIBUTING.md]: Typo: `If`→`It`

Comment (michaelsembwever) [reply CONTRIBUTING.md]: Fixed.  And a small rewording for clarify.

File: NOTICE (hunk 4 above)
Comment (michaelsembwever) [reply NOTICE]: Good catch, thanks.  I got a bit confused bringing all the changes over to https://github.com/apache/cassandra-gocql-driver/pull/1792.  Fixed now.

**Review by martin-sucha — APPROVED** [review]
Looks good to me! Thanks!
